### ISSUE TITLE
Cannot install newer versions from CocoaPods

### ISSUE THREAD

**Issue by squalx** [card-io/card.io-iOS-SDK#219]
I cannot install 5.4.1 (I've tried older ones like 5.3.0) on Xcode 7 or 8, Swift or Objective C projects. As you can see I've tested on several different projects just in case. I can install 5.2.0, but that one is not open source and I need the open source version because I'm creating a Framework that uses CardIO. Since I'm using Dynamic linking, the static framework is giving me some headaches.

Below is the error I get. The installation stalls when it reaches CardIO (in some tests I've done it's the only library being installed) and then I get an error after like a minute o two:
```
[!] Error installing CardIO
[!] /usr/bin/git clone https://github.com/card-io/card.io-iOS-SDK.git /var/folders/8w/bmpj1h_959b1_sdld40chv_h0000gn/T/d20160919-34294-1hmsrjl --single-branch --depth 1 --branch 5.4.1

Cloning into '/var/folders/8w/bmpj1h_959b1_sdld40chv_h0000gn/T/d20160919-34294-1hmsrjl'...
error: RPC failed; curl 56 SSLRead() return error -36
fatal: The remote end hung up unexpectedly
fatal: early EOF
fatal: unpack-objects failed 
```

**Comment by bluk** [comment]
@squalx Can you clone this repository with git by itself?

FYI, the source which the binaries are based on are at https://github.com/card-io/card.io-iOS-source with the appropriate tags for releases. Be sure to follow the [dev setup instructions](https://github.com/card-io/card.io-iOS-source#dev-setup). You may also want to try just building the project via Carthage as documented in the [source repo's README](https://github.com/card-io/card.io-iOS-source#using-cardio) and using the generated framework there depending on your needs.

**Comment by squalx** [comment]
Good call. No, can't download the zipped code, or clone the code. It stalls at 9% or so.

**Comment by webfaqtory** [comment]
I have exactly the same problem. I issue the command "pod install" and get the above error after a few minutes. I have tried editing the Podfile.lock with many different versions of CardIO, all with the same results

**Comment by bluk** [comment]
@webfaqtory Can you clone the repository itself? If there are connection problems, I think the best solution is to contact [GitHub Support](https://github.com/contact) to see if they can help.

**Comment by webfaqtory** [comment]
Yes I clone the repository but other developers will not know how to do this, the will be using "pod install". It is not a connection problem. I have tried on 3 machines, many times, all with the same result. If I do "git clone https://github.com/card-io/card.io-iOS-SDK.git /var/folders/j0/ktzl12bx0vbbt4s441nlr2cw0000gn/T/d20160921-2075-klj4wg --template= --single-branch --depth 1 --branch 5.4.1" it always stops on 47%, every time

**Comment by squalx** [comment]
Doesn't sound like a connection problem. I'm pretty sure the repo is corrupted.

**Comment by bluk** [comment]
I cloned the repository using almost the same parameters (except the directory to put it into):

```
$ git clone https://github.com/card-io/card.io-iOS-SDK.git a --template= --single-branch --depth 1 --branch 5.4.1
Cloning into 'a'...
remote: Counting objects: 59, done.
remote: Compressing objects: 100% (54/54), done.
remote: Total 59 (delta 9), reused 29 (delta 1), pack-reused 0
Unpacking objects: 100% (59/59), done.
Checking connectivity... done.
Note: checking out '12b83d6d6a37250a2fb7b373d7b65a6a5389255d'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

Checking out files: 100% (43/43), done
```

Using:

```
git --version
git version 2.9.3
```

Dunno if a later version of git would help. Can either of you also try a different network if possible to try to clone with? I don't know if the different machines were on a different network.

I went ahead and submitted a question to GitHub support to see if they could help as well.

**Comment by webfaqtory** [comment]
Just tried again on another machine

Johns-Mac:Apps johnburch$ git clone https://github.com/card-io/card.io-iOS-SDK.git /var/folders/j0/ktzl12bx0vbbt4s441nlr2cw0000gn/T/d20160921-2075-klj4wg --template= --single-branch --depth 1 --branch 5.4.1
Cloning into '/var/folders/j0/ktzl12bx0vbbt4s441nlr2cw0000gn/T/d20160921-2075-klj4wg'...
remote: Counting objects: 59, done.
remote: Compressing objects: 100% (54/54), done.
error: RPC failed; curl 56 SSLRead() return error -9806
fatal: The remote end hung up unexpectedly
fatal: early EOF
fatal: unpack-objects failed
Johns-Mac:Apps johnburch$ git --version
git version 2.8.4 (Apple Git-73)

It stuck on 49% for about 10 mins.

**Comment by webfaqtory** [comment]
Just tried 5.2.0 and that worked fine on all 3 machines, took less than a minute. 5.4.1 is stuck again on 49%

**Comment by squalx** [comment]
My version of Git is 2.8.4 too, so that might be the culprit. When unpacking objects it took forever, but with that latest clone command you sent it worked, @bluk.

I'll update Git and will let you know.

Thanks

**Comment by squalx** [comment]
Git 2.10 and still the same issue with cocoa pods. But like I mentioned, I was able to download the repo with the clone command sent previously:
```
[!] Error installing CardIO
[!] /usr/local/bin/git clone https://github.com/card-io/card.io-iOS-SDK.git /var/folders/8w/bmpj1h_959b1_sdld40chv_h0000gn/T/d20160922-3443-1s6kuwv --single-branch --depth 1 --branch 5.4.1

Cloning into '/var/folders/8w/bmpj1h_959b1_sdld40chv_h0000gn/T/d20160922-3443-1s6kuwv'...
error: RPC failed; curl 56 SSLRead() return error -9806
fatal: The remote end hung up unexpectedly
fatal: early EOF
fatal: unpack-objects failed
```

**Comment by cannyboy** [comment]
Using an earlier version seems to work
````
pod 'CardIO', '~> 5.3.2'
````

**Comment by bluk** [comment]
I contacted GitHub Support and they weren't able to reproduce the issue either (they git cloned it themselves) and they did a fsck and it seemed ok on their side. Also, I tried VPNing into a few different locations around the world and was able to install using CocoaPods successfully (slower in some locations than others but everything eventually completed without an error).

Googling a few other related errors to `curl 56 SSLRead() return error -36` it seems that on some git installations, this will fail when the repository becomes too big. Didn't seem there were any applicable solutions though. Ultimately, the best way forward will probably be having a working CocoaPod Podspec for the [source repository](https://github.com/card-io/card.io-iOS-source). It is on a personal backlog but if someone wants to create a PR, that would be greatly appreciated.

**Comment by vincent-coetzee** [comment]
I'm having exactly the same issue. I've have been trying for days to install this version of the pod without success, and I've had this problem with version 5.4.1 in the past. There is definitely something wrong with the repo. This is most unfortunate, because our project is dependent on this library.

-> Installing CardIO (5.4.1)
 > Git download
 > Git download
     $ /usr/bin/git clone https://github.com/card-io/card.io-iOS-SDK.git /var/folders/pl/ncn36wvx3t7c3257gwhxpx3r0000gn/T/d20161108-834-1qfsflo --template=
     --single-branch --depth 1 --branch 5.4.1
     Cloning into '/var/folders/pl/ncn36wvx3t7c3257gwhxpx3r0000gn/T/d20161108-834-1qfsflo'...
     error: RPC failed; curl 56 SSLRead() return error -36
     fatal: The remote end hung up unexpectedly
     fatal: early EOF
     fatal: unpack-objects failed

**Comment by squalx** [comment]
In case this helps anyone, this is what I tried today to get the latest version and it worked:
    pod 'CardIO', :git => "https://github.com/card-io/card.io-iOS-SDK.git", :commit => "12b83d6d6a37250a2fb7b373d7b65a6a5389255d"
